### DISCUSSION TITLE
Can swiss launch .7z files?

### DISCUSSION THREAD

**Discussion by javier26** [emukidid/swiss-gc#778] [Q&A]
Can swiss launch .7z files or should they be uncompressed iso?

**Comment by Extrems** [comment]
This would be unreasonable even on a PC.

**Comment by javier26** [comment]
Why because it takes a few seconds to extract? There are ports of 7z for power pc, so shouldn't it be technically doable? ( just might have to leave x amount of swap space for extraction) Why is it so crazy? Not trying to be an as$, genuinely curious about the technical limitations.

**Reply by Extrems** [reply]
Assuming you want us to extract to disk rather than memory, it would take approximately 25 minutes to do so.

**Comment by LiquidSebbey** [comment]
There is no memory to do such tasks on the gamecube. Extracting to where?

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by javier26** [reply]
To the SD card, with a delete operation after you quit the game. Seems like it would be no different than having a bunch of ISOs. 2GB swap space seems more than enough. But according to Extrems above it would take forever to extract I guess based on CPU /SD read/write speed.

**Reply by Extrems** [reply]
It's not the CPU that's the bottleneck, but I/O.

**Reply by javier26** [reply]
Now I understand the unreasonableness  :) thanks Extrems.

**Comment by LiquidSebbey** [comment]
Make sense for a PowerPC on 485mhz (3x162mhz FSB)

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Extrems** [reply]
Good job quoting a Wikipedia error failing to do basic math.

**Reply by LiquidSebbey** [reply]
I read the CPU speed on https://www.anandtech.com/show/858/2 
Not a wikipedia article. 
But the SD card speed should be the main bottleneck.
Is the speed writing to SD card if different using SP2SD or memorycard slot? Or the same basically?

**Reply by Extrems** [reply]
They're the same, but since we're talking about a 3.375 MB/s bus and we're reading from and writing to the same device, making the operation half-duplex at best, we're looking at something like 1 MB/s accounting for overhead.

On the GC Loader, it would take over an hour and blow a significant chunk of your SD card's write endurance.

**Reply by LiquidSebbey** [reply]
Wow... so, having a SP2SD with the 7z file and another SD card in a memory card as "the buffer" slot would make it "full duplex"... Thinking out of the box: would a SD card connected to the broadband port be a solution? (No, not available, I know) Or is that using the same bus too?

**Reply by Extrems** [reply]
It's the very same. It's even shared with Memory Slot A.

**Reply by LiquidSebbey** [reply]
Oh wow!.... Big N saved pennies there! ;-)

**Reply by Extrems** [reply]
Well, a SPI bus normally has multiple devices. It's odd that they have 3 of them.
